Sources/App/Services/Q17Service.swift

```diff
@@ -0,0 +1,59 @@
+//
+// This source file is part of the ENGAGE-HF-AI-Voice open source project
+//
+// SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+import ModelsR4
+import Vapor
+
+
+/// Service for managing Q17 questionnaire
+enum Q17Service: QuestionnaireService {
+    static let storage = QuestionnaireStorageService(
+        questionnaireName: "q17",
+        filePath: FileService.q17FilePath,
+        directoryPath: FileService.q17DirectoryPath
+    )
+    static let manager = QuestionnaireManager(questionnaire: storage.loadQuestionnaire())
+    
+    
+    static func loadAnsweredQuestions(phoneNumber: String, logger: Logger) {
+        let currentResponse = storage.loadQuestionnaireResponse(phoneNumber: phoneNumber, logger: logger)
+        manager.setCurrentResponse(currentResponse)
```

Comment:
Wouldn't this be shared across all requests, e.g., even if two come in at the same time. Shouldn't we move from static global singletons to services that are created on the creation of the web service and then are destroyed at the end. They could then store the phone number internally or keep a reference to it and could be passed into ones that are managing it? I think the fact that we use singletons and static functions will hurt us here in the long run if we think about any parallelization, e.g., two phone calls at the same time.

Also applies to a lot of other sections of this code.

Reply:
Absolutely - good catch!